### PR TITLE
Fix duplicate messages when using new style tokens

### LINKED ISSUE #128: My own messages are shown twice
Hi. First off thanks for the great app!

I use Quassel IRC and when I send a message a split second later the same message appears again. As if I sent it twice.

Others do not see doubled messages, even I don't see it in native slack client. So it is probably something between Quassel and irc-slack that makes Qassel show my own message 2 times.

### PR DESCRIPTION
**Pull request by insomniacslk**
Fixes #128

Signed-off-by: Andrea Barberio <insomniac@slackware.it>

### CODE REVIEW

**Comment by insomniacslk** [comment]
Tested with
* legacy token. Sending a message from IRC client, and from web client
* new-style token. Sending a message from IRC client, and from web client

In all cases the message shows up only once on all clients

File: event_handler.go [hunk 1]
```diff
@@ -282,6 +282,15 @@ func eventHandler(ctx *IrcContext, rtm *slack.RTM) {
 		case *slack.MessageEvent:
 			// https://api.slack.com/events/message
 			message := ev.Msg
+			if !ctx.usingLegacyToken && message.User == ctx.User.ID && message.ClientMsgID == "" {
```
Comment (janza) [inline event_handler.go:285]: There is a check with similar purpose in [print_message function](https://github.com/insomniacslk/irc-slack/blob/master/event_handler.go#L255). Might be good to consider having both checks in the same place?

Comment (insomniacslk) [reply event_handler.go:285]: that's the code I was looking for and couldn't find, thank you! Yeah I'll move this logic there

Comment (insomniacslk) [reply event_handler.go]: fixed

Comment (insomniacslk) [reply event_handler.go]: not yet working, probably need to remove the bot ID check

Comment (insomniacslk) [reply event_handler.go]: ok, this seems to be working now

**Pull request merged**